### DISCUSSION TITLE
Is it possible to configure semver major version zero changes being considered a major change?

### DISCUSSION THREAD

**Discussion by herndlm** [renovatebot/renovate#13986] [Request Help] [answered]
Sorry for the cryptic title, it's hard to put that in words. But the use case is simple and must be an often-asked question. Unfortunately I did not find an answer so far :/

Manager: composer
Package + version constraint: e.g. `nategood/httpful:^0.2.20`
Renovate considers the update to `^0.3.0` a **minor** update.

I'm not a 100% sure if the terms are correct here, but I would consider this to behave more like a **major** update, also according to the 0.x behaviour described at https://semver.org/#spec-item-4.
The used rangeStragey is currently `update-lockfile`.

The concrete referenced example here is https://github.com/phpstan/phpstan-src/pull/979 which is using the following config https://github.com/phpstan/phpstan-src/blob/master/.github/renovate.json

Is there a way to make Renovate behave here as if this would be a "real" major version bump? I have the feeling that I'm missing something very obvious :) The main problem in that example is probably that this dependency is grouped with other minor updates but for 0.x it's very likely that things break which makes me want to move it in the major group.

**Comment by rarkins** [comment, marked as answer]
See #1257 and https://github.com/renovatebot/renovate/issues/4852

**Reply by andersliland** [reply]
This issue has been deleted. What was the solution?

Is there some native support for this, or some presets? 

I have not been able to find any.

**Reply by rarkins** [reply]
I think it was converted to this discussion: https://github.com/renovatebot/renovate/discussions/21656

In short: "this" is wrong, and we don't intend to support anything which is wrong. Asking to treat non-major changes as major is like asking to treat the month of March as February.

What everyone needs to do is stop asking to treat minor updates as major (which makes no sense) and instead think in terms of breaking/non-breaking concepts in parallel to major/minor/patch. More details within the linked discussion.
